Source/SwiftLintFramework/Models/LinterCache.swift

```diff
@@ -13,12 +13,31 @@ public enum LinterCacheError: Error {
     case invalidFormat
     case differentVersion
     case differentConfiguration
+    case inconsistentLastRunDate
 }
 
 public final class LinterCache {
     private var cache: [String: Any]
     private let lock = NSLock()
 
+    public var lastRunDate: Date? {
+        get {
+            lock.lock()
```

Comment:
Yeah, and also we could use a safe dictionary where the operations used that lock. However, I wanted to reduce the scope of this change.

Also, I though that this particular attribute would be a really nice place to use defer to ensure unlocking the lock